src/ray/common/buffer.h

```diff
@@ -55,9 +55,9 @@ class LocalMemoryBuffer : public Buffer {
       : has_data_copy_(copy_data) {
     if (copy_data) {
       RAY_CHECK(data != nullptr);
-      buffer_.insert(buffer_.end(), data, data + size);
```

Comment:
Is the performance issue caused by the `insert`, instead of constructing a vector? Would it be an issue if we just create an vector dynamically?

Reply:
> Is the performance issue caused by the `insert`, instead of constructing a vector? Would it be an issue if we just create an vector dynamically?

I test creating a vector dynamically instead of insert, I get a very similar call stack, including `std::__uninitialized_copy() ` and `std::_Construct`. So lead to similar performance.
![20191209145035](https://user-images.githubusercontent.com/4771905/70413823-506c2200-1a93-11ea-8e7e-8e746a242f3d.jpg)